src/vs/platform/extensionManagement/common/extensionManagement.ts

```diff
@@ -85,6 +91,7 @@ export interface IExtensionContributions {
 	menus?: { [context: string]: IMenu[] };
 	snippets?: ISnippet[];
 	themes?: ITheme[];
+	explorer?: ITreeExplorer;
```

Comment:
[needs thinking] The problem is that this means no extension can ever add more than one tree-viewlet